DCC++ Current sensing broken

Latest JMRI v4.13.4 testing release, originally discovered in 4.12 release
Latest official DCC++ Git release
Platform: Linux 4.9.59-v7+ on Raspberry Pi

Looks like the parsing of DCC++ current sensing responses is broken.
DCC++ traffic log:
```
[TX: c]   Read Track Current Cmd 
[RX: a0]   Current: 0 / 1024
[TX: 1]   Track Power ON Cmd 
[RX: p1]   Power Status: ON
[TX: c]   Read Track Current Cmd 
[RX: a14]   Current:  / 1024
[TX: c]   Read Track Current Cmd 
[RX: a16]   Current:  / 1024
[TX: 0]   Track Power OFF Cmd 
[RX: p0]   Power Status: OFF
[TX: c]   Read Track Current Cmd 
[RX: a0]   Current: 0 / 1024
```

When the response isn't 0, JMRI throws this error:
```
2018-11-17 23:20:40,392 dccpp.DCCppReply                      ERROR - DCCppReply value index too big. idx = 2 msg = a16 [AWT-EventQueue-0]
2018-11-17 23:20:40,396 dccpp.DCCppReply                      ERROR - DCCppReply value index too big. idx = 2 msg = a16 [AWT-EventQueue-0]
2018-11-17 23:20:40,398 jmrix.AbstractMRTrafficController     ERROR - Unexpected exception in invokeAndWait: {}java.lang.reflect.InvocationTargetException [serial.SerialDCCppPacketizer Receive thread]
java.lang.reflect.InvocationTargetException
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1319)
	at java.awt.EventQueue.invokeAndWait(EventQueue.java:1294)
	at javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1348)
	at jmri.jmrix.AbstractMRTrafficController.distributeReply(AbstractMRTrafficController.java:1013)
	at jmri.jmrix.AbstractMRTrafficController.handleOneIncomingReply(AbstractMRTrafficController.java:1056)
	at jmri.jmrix.AbstractMRTrafficController.receiveLoop(AbstractMRTrafficController.java:868)
	at jmri.jmrix.AbstractMRTrafficController$2.run(AbstractMRTrafficController.java:792)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.parseInt(Integer.java:615)
	at jmri.jmrix.dccpp.DCCppReply.getCurrentInt(DCCppReply.java:768)
	at jmri.jmrix.dccpp.DCCppMultiMeter.message(DCCppMultiMeter.java:39)
	at jmri.jmrix.dccpp.DCCppTrafficController.forwardReply(DCCppTrafficController.java:85)
	at jmri.jmrix.AbstractMRTrafficController.notifyReply(AbstractMRTrafficController.java:295)
	at jmri.jmrix.AbstractMRTrafficController$RcvNotifier.run(AbstractMRTrafficController.java:1223)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

--- a/java/src/jmri/jmrix/dccpp/DCCppConstants.java
+++ b/java/src/jmri/jmrix/dccpp/DCCppConstants.java
@@ -134,8 +134,8 @@ public final class DCCppConstants {
     public final static String LIST_SENSORS_REPLY_REGEX = "\\s*Q\\s*(\\d+)\\s+(\\d+)\\s+([0,1])\\s*";
     public final static String PROGRAM_REPLY_REGEX = "\\s*r\\s*(\\d+)\\|(\\d+)\\|(\\d+)\\s+([-]*\\d+)\\s*";
     public final static String PROGRAM_BIT_REPLY_REGEX = "\\s*r\\s*(\\d+)\\|(\\d+)\\|(\\d+)\\s+(\\d+)\\s+(\\d+)\\s*";
-    public final static String CURRENT_REPLY_REGEX = "\\s*a\\s*(\\d+)";
-    public final static String CURRENT_REPLY_NAMED_REGEX = "\\s*a\\s*(\\w+)\\s*(\\d+)";
+    public final static String CURRENT_REPLY_REGEX = "\\s*a\\s*(\\d+)\\s*";
+    public final static String CURRENT_REPLY_NAMED_REGEX = "\\s*a\\s*(\\w*?[a-zA-Z])\\s*(\\d+)\\s*";
     public final static String TRACK_POWER_REPLY_REGEX = "\\s*p\\s*([0,1])\\s*";
     public final static String TRACK_POWER_REPLY_NAMED_REGEX = "\\s*p\\s*(\\d+)\\s+(\\w+)\\s*";
     public final static String SENSOR_REPLY_REGEX = "\\s*[Qq]\\s*(\\d+)\\s*";

--- a/java/test/jmri/jmrix/dccpp/DCCppReplyTest.java
+++ b/java/test/jmri/jmrix/dccpp/DCCppReplyTest.java
@@ -95,14 +95,40 @@ public class DCCppReplyTest extends jmri.jmrix.AbstractMessageTestBase {
     @Test
     public void testNamedCurrentReply() {
         DCCppReply l = DCCppReply.parseDCCppReply("a MAIN 0");
+        Assert.assertTrue(l.isCurrentReply());
         Assert.assertTrue(l.isNamedCurrentReply());
         Assert.assertEquals('a', l.getOpCodeChar());
         Assert.assertEquals("0", l.getCurrentString());
 
         l = DCCppReply.parseDCCppReply("a MAIN 100");
+        Assert.assertTrue(l.isCurrentReply());
         Assert.assertTrue(l.isNamedCurrentReply());
         Assert.assertEquals('a', l.getOpCodeChar());
         Assert.assertEquals("100", l.getCurrentString());
+
+        l = DCCppReply.parseDCCppReply("aMAIN0");
+        Assert.assertTrue(l.isCurrentReply());
+        Assert.assertTrue(l.isNamedCurrentReply());
+        Assert.assertEquals('a', l.getOpCodeChar());
+        Assert.assertEquals("0", l.getCurrentString());
+
+        l = DCCppReply.parseDCCppReply("aMAIN41");
+        Assert.assertTrue(l.isCurrentReply());
+        Assert.assertTrue(l.isNamedCurrentReply());
+        Assert.assertEquals('a', l.getOpCodeChar());
+        Assert.assertEquals("41", l.getCurrentString());
+
+        l = DCCppReply.parseDCCppReply("a41");
+        Assert.assertTrue(l.isCurrentReply());
+        Assert.assertFalse(l.isNamedCurrentReply());
+        Assert.assertEquals('a', l.getOpCodeChar());
+        Assert.assertEquals("41", l.getCurrentString());
+
+        l = DCCppReply.parseDCCppReply("a 41");
+        Assert.assertTrue(l.isCurrentReply());
+        Assert.assertFalse(l.isNamedCurrentReply());
+        Assert.assertEquals('a', l.getOpCodeChar());
+        Assert.assertEquals("41", l.getCurrentString());
     }
 
     // The minimal setup for log4J

